### PR TITLE
Switch to the AMTL argument parser.

### PR DESCRIPTION
**Pull request by dvander**
This removes a bunch of hand-rolled argument parsing code. Two options
have been removed: -^ and -\ which allowed switching the control
character, generally a poor idea for having shareable code.

### CODE REVIEW

File: compiler/parser.cpp [hunk 1]
```diff
@@ -717,189 +721,182 @@ set_extension(char* filename, const char* extension, int force) {
...
-                    if (*(ptr + 1) != '\0')
-                        about();
-                    sc_asmfile = TRUE; /* skip last pass of making binary file */
-                    if (verbosity > 1)
-                        verbosity = 1;
-                    break;
-                case 'D': /* set active directory */
-                    ptr = option_value(ptr, argv, argc, &arg);
+
+    auto usage = Sprintf("[options] <filename> [filename...]", argv[0]);
```
Comment (peace-maker) [inline compiler/parser.cpp]: That `argv[0]` argument seems unused.

File: compiler/parser.cpp [hunk 2]
```diff
@@ -717,189 +721,182 @@ set_extension(char* filename, const char* extension, int force) {
...
+    if (opt_error_file.hasValue())
+        strlcpy(ename, opt_error_file.value().chars(), _MAX_PATH);
+
 #if defined __WIN32__ || defined _WIN32 || defined _Windows
-                case 'H':
-                    hwndFinish = (HWND)atoi(option_value(ptr, argv, argc, &arg));
-                    if (!IsWindow(hwndFinish))
-                        hwndFinish = (HWND)0;
-                    break;
+    hwndFinish = (HWND)atoi(opt_hwnd.value().chars());
```
Comment (peace-maker) [inline compiler/parser.cpp]: This crashes on windows when not passing the `-H` option due to not having a default value.

File: compiler/parser.cpp [hunk 3]
```diff
@@ -1069,60 +1067,6 @@ setcaption(void) {
     pc_printf("Copyright (c) 2004-2018 AlliedModders LLC\n\n");
 }
 
-static void
-about(void) {
-    if (strlen(errfname) == 0) {
-        setcaption();
-        pc_printf("Usage:   spcomp <filename> [filename...] [options]\n\n");
```
Comment (peace-maker) [inline compiler/parser.cpp]: The old documentation has filenames first then options. The new usage output shows options first. I guess the old way around still works too?

File: compiler/parser.cpp [hunk 4]
```diff
@@ -1069,60 +1067,6 @@ setcaption(void) {
...
-        pc_printf("         -v<num>  verbosity level; 0=quiet, 1=normal, 2=verbose (default=%d)\n",
-                  verbosity);
-        pc_printf("         -w<num>  disable a specific warning by its number\n");
-        pc_printf("         -z<num>  compression level, default=9 (0=none, 1=worst, 9=best)\n");
-        pc_printf("         -E       treat warnings as errors\n");
-        pc_printf("         -\\       use '\\' for escape characters\n");
-        pc_printf("         -^       use '^' for escape characters\n");
-        pc_printf("         -;<+/->  require a semicolon to end each statement (default=%c)\n",
-                  sc_needsemicolon ? '+' : '-');
-        pc_printf("         sym=val  define constant \"sym\" with value \"val\"\n");
```
Comment (peace-maker) [inline compiler/parser.cpp]: All this extra documentation is gone.

File: compiler/parser.cpp (hunk 3 above)
Comment (dvander) [reply compiler/parser.cpp]: Yes.

File: compiler/parser.cpp (hunk 4 above)
Comment (dvander) [reply compiler/parser.cpp]: I can add notes back for sym=val line.

File: compiler/parser.cpp (hunk 2 above)
Comment (dvander) [reply compiler/parser.cpp]: Thanks, good catch, will fix this.

**Comment by dvander** [comment]
Is there any value to the "response file" thing? If not, I'd rather just remove it.

**Comment by dvander** [comment]
Response files are removed.

**Pull request merged**